File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        r = []
        rank=1
        d={}
        c = sorted(arr)
        for i in c :
            if i not in d:
                d[i] = rank
                rank += 1
        for i in arr:
            r.append(d[i])
        return r
                
        

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        int n = arr.size(); 
        priority_queue<pair<int,int>,vector<pair<int,int>>, greater<pair<int,int>>>pq ; 
        for(int i = 0 ; i < n ; i++){
            pq.push({arr[i] , i}) ; 
        }
        int rank =1 ; 
        while(!pq.empty()){
            int num = pq.top().first ; 
            int idx = pq.top().second ; 
            pq.pop()  ; 
            int next = pq.top().first ; 
            arr[idx] = rank ; 
            if(num!=next) rank++ ;
        }
        return arr ; 
    }
};